### PR TITLE
Add adapter for converting LORIS username/password to HTTP Bearer token

### PR DESCRIPTION
**Pull request by driusan**
Added an adapter which converts usernames and passwords to tokens via the LORIS API.

This is not thread-safe, since it uses a hack of adding url as a property to the _loris_adapter function in order to pass the URL between the composite credential and the adapter.

### CODE REVIEW

**Comment by yarikoptic** [comment]
Let me digest it later while looking at the code not in the phone. May be I will see where in the interactions credential could become aware of URL

**Comment by driusan** [comment]
Moving the adapters from bare functions to class methods would make sense to me, but I don't know anything about the other NDA adapter and couldn't test it if I changed that part of the architecture

**Comment by yarikoptic** [comment]
NDA adapter isn't used in production yet, was more of an unfinished hackathon effort,,, so it would be ok to do what needs to be done

**Comment by driusan** [comment]
I realized that it doesn't need to be a method, it just needs to be passed the parent class instance, so I did that instead.. seems more reasonable to me now.

File: datalad/support/third/loris_token_generator.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import sys
```
Comment (yarikoptic) [inline datalad/support/third/loris_token_generator.py:1]: Please borrow the header from other files, or add a custom one (about license/copyright, if needed)

**Review by yarikoptic — APPROVED** [review]
Lovely, especially if it works ;-)

In principle/ideally there should be a test for the interaction.  httpretty might come handy - there are some tests using it.  Since otherwise I am afraid we might break it sometime without realizing that

**Comment by driusan** [comment]
squashed and pushed directly to the branch. Still needs test.

**Pull request closed without merging**